src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -316,11 +316,11 @@ private void OnServerGoAway(long lastProcessedStreamId)
                     return;
                 }
 
-                _lastProcessedStreamId = lastProcessedStreamId;
+                _firstRejectedStreamId = firstRejectedStreamId;
 
                 foreach (KeyValuePair<QuicStream, Http3RequestStream> request in _activeRequests)
                 {
-                    if (request.Value.StreamId > lastProcessedStreamId)
+                    if (request.Value.StreamId >= firstRejectedStreamId)
                     {
                         streamsToGoAway.Add(request.Value);
                     }
```

Comment:
From RFC:

>    *  Upon receipt of a GOAWAY frame, if the client has already sent
      requests with a Stream ID greater than or equal to the identifier
      contained in the GOAWAY frame, those requests will not be
      processed.  Clients can safely retry unprocessed requests on a
      different HTTP connection.  A client that is unable to retry
      requests loses all requests that are in flight when the server
      closes the connection.

Reply:
Good catch @rzikm !